src/core/term.ml

```diff
@@ -379,11 +379,21 @@ type tevar = term_env Bindlib.var
 (** A short name for the type of a boxed {!type:term_env}. *)
 type tebox = term_env Bindlib.box
 
-(** [mkfree x] injects the [Bindlib] variable [x] in a term. *)
-let mkfree : tvar -> term = fun x -> Vari(x)
+(** [of_tvar x] injects the [Bindlib] variable [x] in a term. *)
+let of_tvar : tvar -> term = fun x -> Vari(x)
 
-(** [te_mkfree x] injects the [Bindlib] variable [x] in a {!type:term_env}. *)
-let te_mkfree : tevar -> term_env = fun x -> TE_Vari(x)
+(** [new_tvar s] creates a new [tvar] of name [s]. *)
+let new_tvar : string -> tvar = Bindlib.new_var of_tvar
+
+(** [new_tvar_ind s i] creates a new [tvar] of name [s ^ string_of_int i]. *)
+let new_tvar_ind : string -> int -> tvar = fun s i ->
+  new_tvar (Printf.sprintf "%s%i" s i)
```

Comment:
What do you think of adding the integer as an optional argument to `new_tvar`?

Reply:
What do you have in mind? Where would you use this?

Reply:
Having `new_tvar: string -> ?ind:int -> tvar`

Reply:
I understand that you want to factorize but these two functions are used differently: new_tvar_ind is used in conjunction with Array.init.